packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -101,26 +101,29 @@ library LibProposing {
         // depend on Taiko DAO to make necessary adjustments to the rewards.
         uint256 reward;
 
-        if (config.proposerRewardPerSecond > 0 && config.proposerRewardMax > 0)
-        {
-            // Unchecked is safe as block.timestamp is always greater than
-            // block.proposedAt (proposed in the past)
-            unchecked {
-                uint256 blockTime = block.timestamp
-                    - state.blocks[(b.numBlocks - 1) % config.blockRingBufferSize]
-                        .proposedAt;
-
-                if (blockTime > 0) {
-                    reward = (config.proposerRewardPerSecond * blockTime).min(
-                        config.proposerRewardMax
-                    );
-
-                    // Reward must be minted
-                    LibTaikoToken.creditToken(
-                        state, resolver, msg.sender, reward, true
-                    );
-                }
+        // First proposer per L1 block updates the L1 block height and the accumulated reward.
+        if(state.slotC.latestRewardAccL1Height < block.number) {
+            // The very first proposal within the protocol starts the 'clock' and from that point on we calculate the accumulation.
+            if(state.slotC.latestRewardAccL1Height != 0) {
```

Comment:
How about use 'lastProposedHeight' to represent the L1 block number in which the last block proposal occurred. 

Later we can do:
```
if (proposerRewardPerL1Block > 0) {
    if (block.number != state.slotC.lastProposedHeight) {
       state.slotC.accProposerReward += config.proposerRewardPerL1Block * (block.numeber - state.slotC.lastProposedHeight);
      if (state.slotC.accProposerReward > proposerRewardMax) state.slotC.accProposerReward = proposerRewardMax;
       state.slotC.lastProposedHeight = block.number;
    }

    reward = state.slotC.accProposerReward/3;
    state.slotC.accProposerReward -= reward;
}
```

Reply:
`latestRewardAccL1Height` is the latest L1 height the reward was pooled into the `accumulatedReward`. (It always happens in the first `proposeBlock()` per given L1 block. You want me to change the name ? I can do that, it indeed might be confusing.